InfiniteRecharge/src/main/java/frc/robot/subsystem/scoring/shooter/ball_management/BallManagementSubsystem.java

```diff
@@ -60,11 +61,15 @@ public void periodic(float deltaTime) {
      * @param rate How rapidly should we give balls to the feeder?
      */
     public void fire(float rate) {
-        motor.set(ControlMode.PercentOutput, rate);
-
-        // Because you can neva 'ave enuf dakka!
-        SmartDashboard.putString(getName() + "/State",
-                "DAKKADAKKADAKKADAKKADAKKDAKKADAKKADAKKADAKKADAKKADAKKADAKKDAKKADAKKADAKKADAKKADAKKADAKKADAKKDAKKADAKKADAKKADAKKADAKKADAKKADAKKDAKKADAKKA");
+        if (ShooterSubsystem.upToSpeed == true) {
```

Comment:
You should move this logic into the ShooterSubsystem.fire() function. that way the BMS doesn't need to know about the shooter subsystem at all. We only call this fire() function from the ShooterSubsystem fire() function.